src/core/CorePool.ts

```diff
@@ -325,12 +409,56 @@ export class CorePool {
     tickUpper: number,
     liquidityDelta: JSBI
   ): Position {
-    // TODO
-    return new Position();
+    let _position: Position = this.getPosition(owner, tickLower, tickUpper);
+
+    let _tick = this.tickManager.getTickAndInitIfAbsent(this._tickCurrent);
+
+    let flippedLower = _tick.update(
```

Comment:
It's necessary to add a condition expression like `if (JSBI.notEqual(liquidityDelta, ZERO))` outside these two updates. They are for liquidity maintenance. And fee income update is executed from line 434 to 446 whatever liquidityDelta is given.

Reply:
That's right, we only need these piece of code run when `liquidityDelta` is NOT zero. Nice review!